packages/flutter/lib/src/widgets/basic.dart

```diff
@@ -1668,6 +1668,22 @@ class Positioned extends ParentDataWidget<Stack> {
        bottom = null,
        super(key: key, child: child);
 
+  /// Creates a Positioned object with the values from the given [RelativeRect].
+  ///
+  /// This sets the [left], [top], [right], and [bottom] properties from the
+  /// given [RelativeRect]. The [height] and [width] properties are set to null.
+  Positioned.fromRelativeRect({
+    Key key,
+    Widget child,
+    RelativeRect rect,
+  }) : left = rect.left,
+       top = rect.top,
+       right = rect.right,
+       bottom = rect.bottom,
+       width = null,
+       height = null,
+       super(key: key, child: child);
+
```

Comment:
You could use rect?.left etc here and then assert rect != null. Maybe that's an infinitesimally better failure mode :-).